doc/specs/vulkan/chapters/memory.txt

```diff
@@ -552,58 +552,57 @@ There must: be at least one memory type with the
 ename:VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT bit set in its
 pname:propertyFlags.
 
-The memory types are sorted according to a preorder which serves to aid in
-easily selecting an appropriate memory type.
-Given two memory types X and Y, the preorder defines [eq]#X {leq} Y# if:
-
-  * the memory property bits set for X are a strict subset of the memory
-    property bits set for Y.
-    Or,
-  * the memory property bits set for X are the same as the memory property
-    bits set for Y, and X uses a memory heap with greater or equal
-    performance (as determined in an implementation-specific manner).
-
-Memory types are ordered in the list such that X is assigned a lesser
-pname:memoryTypeIndex than Y if [eq]#(X {leq} Y) {land} {lnot} (Y {leq} X)#
-according to the preorder.
-Note that the list of all allowed memory property flag combinations above
-satisfies this preorder, but other orders would as well.
-The goal of this ordering is to enable applications to use a simple search
-loop in selecting the proper memory type, along the lines of:
+The array pname:memoryTypes must: be ordered in such a way that for each of
+its distinct memory types X and Y it holds that X occupies a lower index
+position than Y if:
+
+  * either the pname:propertyFlags of X (interpreted as a set of raised flags) is a
+    strict subset of the pname:propertyFlags of Y.
+  * or the pname:propertyFlags of X and Y are equal, and X belongs to a memory
+    heap with greater performance (as determined in an
+    implementation-specific manner).
+
+[NOTE]
+.Note
+====
+There might be more than one conformant way to order given set of memory
+types. Notice that the list of all allowed memory property flag
+combinations above is written in one such of several possible ways that do
+satisfy the prescribed order.
+====
```

Comment:
I'm not sure what this note is attempting to clarify. Your new spec text makes the situation a lot clearer - the only reordering that can take place is that of memory types with equivalent propertyFlags. This note seems to be suggesting there are other options and actually makes things less clear IMO... unless I'm missing something?

Reply:
It tries to say the order is not unique. I.e. the same list of memory properties can be ordered validly more than one way.
The utility is to warn applications against expecting exactly the order in the list above.

Reply:
Oh because if for two sets of memory types and one is not a subset of the other, then the ordering between them is arbitrary? Right?

If that's it, maybe you could call that out in this note? Think that would help.